src/components/Dashboard/FilingHistoryList.vue

```diff
@@ -374,1042 +376,665 @@
-        const documents = this.historyItems[i].documents
-        // NB: this only works if there is a filing document
-        if (documents?.length > 0 && documents[0].filingId === filingId) {
-          this.panel = i
-          break
+        if (type === DocumentTypes.RECEIPT) {
+          await this.fetchOneReceipt(item.documents[i]).catch(error => {
+            // eslint-disable-next-line no-console
+            console.log('fetchOneReceipt() error =', error)
+            this.downloadErrorDialog = true
+          })
         }
       }
-    },
-
-    async correctThisFiling (item: HistoryItemIF) {
-      switch (item?.filingType) {
-        case FilingTypes.ANNUAL_REPORT:
-          // FUTURE:
-          // this.$router.push({ name: Routes.ANNUAL_REPORT,
-          //   params: { filingId: filing.filingId, isCorrection: true }})
-          // FOR NOW:
-          this.$router.push({
-            name: Routes.CORRECTION,
-            params: { correctedFilingId: item.filingId }
-          })
-          break
-
-        case FilingTypes.CHANGE_OF_DIRECTORS:
-          // FUTURE:
-          // this.$router.push({ name: Routes.STANDALONE_DIRECTORS,
-          //   params: { filingId: filing.filingId, isCorrection: true } })
-          // FOR NOW:
-          this.$router.push({
-            name: Routes.CORRECTION,
-            params: { correctedFilingId: item.filingId }
-          })
-          break
-
-        case FilingTypes.CHANGE_OF_ADDRESS:
-          // FUTURE:
-          // this.$router.push({ name: Routes.STANDALONE_ADDRESSES,
-          //   params: { filingId: filing.filingId, isCorrection: true } })
-          // FOR NOW:
-          this.$router.push({
-            name: Routes.CORRECTION,
-            params: { correctedFilingId: item.filingId }
-          })
-          break
+    }
+    this.loadingAll = false
+  }
 
-        case FilingTypes.INCORPORATION_APPLICATION:
-          try {
-            // show spinner since the network calls below can take a few seconds
-            this.$root.$emit('showSpinner', true)
+  private showCommentDialog (filingId: number): void {
+    this.currentFilingId = filingId
+    this.addCommentDialog = true
+  }
 
-            // fetch original IA
-            const iaFiling = await this.fetchFilingById(this.entityIncNo, item.filingId)
+  private async hideCommentDialog (needReload: boolean): Promise<void> {
+    this.addCommentDialog = false
+    // if needed, reload comments for this filing
+    // NB: no spinner or state change, just do it quietly // TODO: remove comment if keeping spinner
+    if (needReload) {
+      // find the filing in the list
+      const item = this.historyItems.find(item => (item.filingId === this.currentFilingId))
 
-            if (!iaFiling) {
-              throw new Error('Invalid API response')
-            }
+      // TODO: test "isBusy" spinner when we have commentsLink
+      if (item && item.commentsLink) {
+        this.isBusy = true // TODO: test this
```

Comment:
Yes, I will go through all TODOs before my final submit. Some of this functionality is still blocked by API work.